lib/ecto/adapters/mnesia.ex

```diff
@@ -150,34 +155,44 @@ defmodule Ecto.Adapters.Mnesia do
     end
   end
 
-  @doc """
-  Insert Ecto Schema struct to Mnesia database.
-
-  TODO:
-  - Process `opts`.
-  - Process `on_conflict`
-  - Process `returning`
-  """
-  def insert(_repo, %{autogenerate_id: {pk_field, _pk_type}, schema: schema, source: {_, table}}, params,
+  @doc false
+  # Insert Ecto Schema struct to Mnesia database.
+  # TODO: deal with `opts`: `on_conflict` and `returning`
+  def insert(_repo, %{autogenerate_id: autogenerate_id, schema: schema, source: {_, table}}, params,
              _on_conflict, _returning, _opts) do
-    do_insert(table, schema, pk_field, params)
+    do_insert(table, schema, autogenerate_id, params)
   end
 
-  def insert_all(_repo, %{autogenerate_id: {pk_field, _pk_type}, schema: schema, source: {_, table}},
+  @doc false
+  # Insert all
+  def insert_all(_repo, %{autogenerate_id: autogenerate_id, schema: schema, source: {_, table}},
                  _header, rows, _on_conflict, _returning, _opts) do
     table = table |> Table.get_name()
     count = Table.transaction(fn ->
       rows
       |> Enum.reduce(0, fn params, acc ->
-        do_insert(table, schema, pk_field, params)
+        do_insert(table, schema, autogenerate_id, params)
         acc + 1
       end)
     end)
 
     {count, nil}
   end
 
-  defp do_insert(table, schema, pk_field, params) do
+  # Insert schema without primary keys
+  defp do_insert(table, schema, nil, params) do
+    record = schema |> Record.new(params, table)
+    case Table.insert(table, record) do
+      {:ok, ^record} ->
+        {:ok, params}
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  # Insert schema with auto-generating primary key value
+  # TODO: Check that PK is unique, don't override record
```

Comment:
Found a TODO tag in a comment: # TODO: Check that PK is unique, don't override record